build_darwin.sh

```diff
@@ -1,15 +1,47 @@
 #!/bin/sh
 set -eu
 
-curl -fsSL "https://www.torproject.org/dist/tor-0.3.2.10.tar.gz" -o tor-0.3.2.10.tar.gz
-curl -fsSL "https://www.torproject.org/dist/tor-0.3.2.10.tar.gz.asc" -o tor-0.3.2.10.tar.gz.asc
+curl -fsSL "https://www.openssl.org/source/openssl-$OPENSSL_VERSION.tar.gz" -o openssl-$OPENSSL_VERSION.tar.gz && \
+echo "8c6ff15ec6b319b50788f42c7abc2890c08ba5a1cdcd3810eb9092deada37b0f  openssl-$OPENSSL_VERSION.tar.gz" | shasum -a 256 -c - && \
+tar -xvzf openssl-$OPENSSL_VERSION.tar.gz && \
+cd openssl-$OPENSSL_VERSION && \
+./config --prefix=$PWD/install no-shared no-dso && \
+make
```

Comment:
can we add `make test` for openssl?

Reply:
`1.0m` fails `make test`. `1.0o` is not linkable so disabled that for now. I'm debugging on how to address this issue.

```
../certs/demo/pca-cert.pem: C = AU, ST = Queensland, O = CryptSoft Pty Ltd, CN = Test PCA (1024 bit)
error 18 at 0 depth lookup:self signed certificate
C = AU, ST = Queensland, O = CryptSoft Pty Ltd, CN = Test PCA (1024 bit)
error 10 at 0 depth lookup:certificate has expired
OK
make[1]: *** [test_verify] Error 2
make: *** [tests] Error 2
```

Reply:
how is it on Linux?

Reply:
`make test` appears to pass for OpenSSL on Linux.  I don't know what the discrepancy is offhand.